module/storage.rb

```diff
@@ -0,0 +1,108 @@
+require 'json'
+require_relative '../lib/book'
+require_relative '../lib/label'
+require_relative '../lib/genre'
+require_relative '../lib/game'
+require_relative '../lib/author'
+
+module Storage
+  def save_books(books)
+    save_data('./data/book.json', array_to_hash(books))
+  end
+
+  def save_albums(albums)
+    save_data('./data/album.json', array_to_hash(albums))
+  end
+
+  def save_games(games)
+    save_data('./data/game.json', array_to_hash(games))
+  end
+
+  def save_extra_details(label, genre, author)
+    save_data('./data/genre.json', array_to_hash(genre))
+    save_data('./data/author.json', array_to_hash(author))
+    save_data('./data/label.json', array_to_hash(label))
+  end
+
+  def fetch_books
+    books = fetch_data('./data/book.json')
+    book_arr = []
+    books&.each do |book|
+      book_sample = Book.new(book['publisher'], book['cover_state'], book['publish_date'])
+      extra_to_array(book_sample, book)
+      book_arr << book_sample
+    end
+    book_arr
+  end
+
+  def fetch_labels
+    labels = fetch_data('./data/label.json')
+    labels&.map do |label|
+      Label.new(label['title'], label['color'])
+    end
+  end
+
+  def fetch_albums
+    albums = fetch_data('./data/album.json')
+    albums_arr = []
+    albums&.each do |album|
+      album_sample = MusicAlbum.new(album['on_spotify'], album['publish_date'])
+      extra_to_array(album_sample, album)
+      albums_arr << album_sample
+    end
+
+    albums_arr
+  end
+
+  def fetch_genres
+    genres = fetch_data('./data/genre.json')
+    genres&.map do |genre|
+      Genre.new(genre['name'])
+    end
+  end
```

Comment:
- [OPTIONAL] It would be nice to create the data directory and the JSON files if they are not present in the root directory. Currently, you are accessing them in the root directory because you created them. If they are deleted, your app will crash and you would have to create them manually. I would advise that you safely create the directory and JSON files if they are not present in the root directory. 👍 😎